libmcount/mcount.c

```diff
@@ -1408,8 +1436,16 @@ static void cygprof_exit(unsigned long parent, unsigned long child)
 	mtdp->idx--;
 }
 
-void xray_entry(unsigned long parent, unsigned long child,
-		struct mcount_regs *regs)
+static void cygprof_exit(unsigned long parent, unsigned long child)
+{
+	int saved_errno = errno;
+
+	__cygprof_exit(parent, child);
+	errno = saved_errno;
+}
+
+static void _xray_entry(unsigned long parent, unsigned long child,
+			struct mcount_regs *regs)
```

Comment:
Did you use a single underline to avoid the conflict with `__xray_entry` in `arch/x86_64/xray.S`?